### PR TITLE
test: Clean up inheritance of TestMachines

### PR DESCRIPTION
**Pull request by martinpitt**
Since commit 70894926b7, TestMachines does not need `NetworkCase` any
more, so the inheritance hacks can be dropped. Eventually we want to
move addIface() into a more generic class, but that should happen
similar to StorageHelpers.

### CODE REVIEW

**Comment by martinpitt** [comment]
@marusak: Indeed, I just sent PR #13742 and wondered why we never spotted that before. And of course, due to the thing that this PR fixes we never called any cleanup, and thus did not check for unexpected messages either. So maybe it makes more sense to merge the PRs, as #13742 will have a hard time to succeed without this?

**Comment by martinpitt** [comment]
Merged #13742 here, as either will have a hard time to succeed without the other..

**Comment by martinpitt** [comment]
There are several more unexpected messages:

 - [debian-stable](https://logs.cockpit-project.org/logs/pull-13741-20200317-205820-2610dbdd-debian-stable/log.html), [centos-8-stream](https://logs.cockpit-project.org/logs/pull-13741-20200317-205820-2610dbdd-centos-8-stream/log.html), [ubuntu-stable](https://logs.cockpit-project.org/logs/pull-13741-20200317-205820-2610dbdd-ubuntu-stable/log.html#77) : "error: Tried changing state of a disconnected RFB object"
 - [rhel-8-2](https://logs.cockpit-project.org/logs/pull-13741-20200317-210139-2610dbdd-rhel-8-2/log.html), [centos-8-stream](https://logs.cockpit-project.org/logs/pull-13741-20200317-205820-2610dbdd-centos-8-stream/log.html#81) : `error: Failed when connecting: Connection closed (code: 1000)`
 - [centos-8-stream](https://logs.cockpit-project.org/logs/pull-13741-20200317-205820-2610dbdd-centos-8-stream/log.html#81-2), [ubuntu-stable](https://logs.cockpit-project.org/logs/pull-13741-20200317-205820-2610dbdd-ubuntu-stable/log.html#77): `couldn't shutdown fd: Transport endpoint is not connected`
 - [debian-stable](https://logs.cockpit-project.org/logs/pull-13741-20200317-205820-2610dbdd-debian-stable/log.html): `apparmor="DENIED" operation="capable" profile="/usr/sbin/libvirtd" pid=10995 comm="libvirt_parthel" capability=17  capname="sys_rawio"`
 -  [ubuntu-2004](https://logs.cockpit-project.org/logs/pull-13741-20200317-205820-2610dbdd-ubuntu-2004/log.html#79-2):

         apparmor="DENIED" operation="open" profile="libvirt-888b0127-eee0-4a69-bd6a-7933c236f2ed" name="/" pid=33609 comm="qemu-system-x86" requested_mask="r" denied_mask="r" fsuid=64055 ouid=0
        apparmor="DENIED" operation="open" profile="libvirt-888b0127-eee0-4a69-bd6a-7933c236f2ed" name="/sys/bus/nd/devices/" pid=33609 comm="qemu-system-x86" requested_mask="r" denied_mask="r" fsuid=64055 ouid=0

 - [ubuntu-stable](https://logs.cockpit-project.org/logs/pull-13741-20200317-205820-2610dbdd-ubuntu-stable/log.html#77): `Process 18324 (qemu-system-x86) of user 64055 dumped core.`

 - [fedora-31](https://logs.cockpit-project.org/logs/pull-13741-20200317-205820-2610dbdd-fedora-31/log.html#77-2) (due to React errors in debug mode): `error: Warning: Each child in a list should have a unique "key" prop`

There's probably some more, but it's hard to keep track.

**Comment by martinpitt** [comment]
OK, I think I caught all (or at least most) unexpected messages. Wash, rinse, repeat! (Ugh @ technical debt..)

**Comment by marusak** [comment]
A few more:
- `127.0.0.1:5901: couldn't shutdown fd: Transport endpoint is not connected`
- `audit: type=1400 audit(1584487308.478:83): apparmor="DENIED" operation="open" profile="libvirt-364cd049-96ba-47a2-8c43-94cbfd361f00" name="/mnt/libvirt/vm_one/mydiskofpoolone_temporary" pid=49479 comm="qemu-system-x86" requested_mask="r" denied_mask="r" fsuid=64055 ouid=64055`
- `testlib.Error: error: Warning: An update (setState, replaceState, or forceUpdate) was scheduled from inside an update function. Update functions should be pure, with zero side-effects. Consider using componentDidUpdate or a callback.`
- `Failed to get COMM: No such process`
- `Process 26846 (qemu-system-x86) of user 64055 dumped core.`
- `audit: type=1400 audit(1584491149.410:1793): apparmor="DENIED" operation="capable" profile="libvirtd" pid=19899 comm="libvirt_parthel" capability=17  capname="sys_rawio"`

- `Process 7143 (ip6tables-resto) of user 0 dumped core.` (Unrelated to machines though :) [debian-testing](https://logs.cockpit-project.org/logs/pull-13741-20200317-225449-07ad1ecd-debian-testing/log.html#117-2) )

File: test/verify/check-machines-dbus [hunk 1]
```diff
@@ -371,6 +371,10 @@ class TestMachinesDBus(machineslib.TestMachines):
         b.click("#vm-VmNotInstalled-networks")
         b.wait_present("#vm-VmNotInstalled-network-1-type")
 
+        self.allow_browser_errors("Failed when connecting: Connection closed")
+        self.allow_browser_errors("Tried changing state of a disconnected RFB object")
+        self.allow_journal_messages("couldn't shutdown fd: Transport endpoint is not connected")
```
Comment (marusak) [inline test/verify/check-machines-dbus]: this needs `.*` at the beginning

File: test/verify/machineslib.py [hunk 2]
```diff
@@ -196,10 +194,18 @@ def setUp(self):
...
-        pass
+        # FIXME: report downstream; AppArmor noisily denies some operations, but they are not required for us
+        self.allow_journal_messages('.* type=1400 .* apparmor="DENIED" operation="capable" profile="/usr/sbin/libvirtd.* capname="sys_rawio" .*')
+        if m.image in ["ubuntu-2004"]:
+            self.allow_journal_messages('.* type=1400 .* apparmor="DENIED" operation="open" profile="libvirt.* name="/" .* denied_mask="r" .*')
+            self.allow_journal_messages('.* type=1400 .* apparmor="DENIED" operation="open" profile="libvirt.* name="/sys/bus/nd/devices/" .* denied_mask="r" .*')
+
+        # FIXME: report downstream: qemu often crashes in testAddDisk and testMultipleSettings
+        if m.image in ["ubuntu-stable"]:
+            self.allow_journal_messages('Process .* (qemu-system-x86) of user .* dumped core.')
```
Comment (marusak) [inline test/verify/machineslib.py]: This [seems](https://logs.cockpit-project.org/logs/pull-13741-20200317-225449-07ad1ecd-ubuntu-stable/log.html#244-2) not to get applied. My guess are those brackets.

File: test/verify/machineslib.py [hunk 3]
```diff
@@ -1040,6 +1046,10 @@ def release_target(target):
         # avoid error noise about resources getting cleaned up
         b.logout()
 
+        # AppArmor doesn't like the non-standard path for our storage pools
+        if m.image in ["debian-testing"]:
+            self.allow_journal_messages('.* type=1400 .* apparmor="DENIED" operation="open" profile="libvirtd.* name="%s .*denied_mask="r"' % self.tmp_storage)
```
Comment (marusak) [inline test/verify/machineslib.py]: Not sure if there are more similar problems, or this needs to be more robust, but it does not cover:
```
audit: type=1400 audit(1584487308.478:83): apparmor="DENIED" operation="open" profile="libvirt-364cd049-96ba-47a2-8c43-94cbfd361f00" name="/mnt/libvirt/vm_one/mydiskofpoolone_temporary" pid=49479 comm="qemu-system-x86" requested_mask="r" denied_mask="r" fsuid=64055 ouid=64055
```

File: test/verify/machineslib.py [hunk 4]
```diff
@@ -196,10 +194,18 @@ def setUp(self):
         # we don't have configuration to open the firewall for local libvirt machines, so just stop firewalld
         m.execute("systemctl stop firewalld; systemctl try-restart libvirtd")
 
-    def tearDown(self):
-        # FIXME: Call `super.tearDown()` or remove this method. There are some unexpected messages
-        # which need to be adjusted before we can do that.
-        pass
+        # FIXME: report downstream; AppArmor noisily denies some operations, but they are not required for us
+        self.allow_journal_messages('.* type=1400 .* apparmor="DENIED" operation="capable" profile="/usr/sbin/libvirtd.* capname="sys_rawio" .*')
```
Comment (marusak) [inline test/verify/machineslib.py]: This is not robust enough, does not match:
```
audit: type=1400 audit(1584491149.410:1793): apparmor="DENIED" operation="capable" profile="libvirtd" pid=19899 comm="libvirt_parthel" capability=17 capname="sys_rawio"
```

**Comment by marusak** [comment]
@martinpitt I updated that commit and pushed here as it can hopefully finish the run before the rest of the team gets to the work :)
See [diff](https://github.com/cockpit-project/cockpit/compare/07ad1ecde4fa32357ae1f0a9d3e5dbe6a4fd59b8..6f036ce1615228bc26fce434f7e303e07b90a367) for my updates. I hope I did not break something badly :)

**Comment by marusak** [comment]
In fedora-31 we noticed, that we still don't call cleanup when some methods raise exception.
Simple reproducer:
```
diff --git test/common/testlib.py test/common/testlib.py
index c3a352a80..a3219e20e 100644
--- test/common/testlib.py
+++ test/common/testlib.py
@@ -1147,6 +1147,7 @@ class MachineCase(unittest.TestCase):
             title: Used for the filename.
         """
         if self.browser is not None:
+            raise Exception("I don't know")
             self.browser.snapshot(title, label)
 
     def copy_js_log(self, title, label=None):
diff --git test/verify/check-machines-dbus test/verify/check-machines-dbus
index 33fe88230..e33d2f035 100755
--- test/verify/check-machines-dbus
+++ test/verify/check-machines-dbus
@@ -592,7 +592,7 @@ class TestMachinesDBus(machineslib.TestMachines):
         b.wait_present("#vm-subVmTest1-disks-vdc-device")
         b.click("#delete-subVmTest1-disk-vdc")
         b.wait_present(".modal-dialog")
-        b.click(".modal-footer button:contains(Remove)")
+        b.click(".modal-footer button:contains(Lemove)")
         b.wait_present(".modal-footer .spinner")
         b.wait_not_present("#vm-subVmTest1-disks-vdc-device")
         b.wait_not_present(".modal-dialog")
```

**Comment by martinpitt** [comment]
[One more browser error](https://logs.cockpit-project.org/logs/pull-13741-20200318-054918-6f036ce1-fedora-31/log.html#244): error: Warning: Failed prop type: The prop `libvirtVersion` is marked as required in `HostVmsList`, but its value is `undefined`.

I. e. this is yet another bug on the Machines page, but let's (continue to) ignore it for now.

**Comment by martinpitt** [comment]
[fedora-32](https://logs.cockpit-project.org/logs/pull-13741-20200318-074051-bdba868b-fedora-32/log.html#201) is a bit messy, but for an unrelated reason -- I sent a fix to PR #13744. The rest looks promising!

**Comment by martinpitt** [comment]
[ubuntu-stable](https://logs.cockpit-project.org/logs/pull-13741-20200318-074051-bdba868b-ubuntu-stable/log.html#244) has a message which we don't catch yet:

    audit: type=1400 audit(1584519622.866:711): apparmor="DENIED" operation="open" profile="libvirt-6c2a01b4-17bd-4e8c-b3fb-5e623e866fbb" name="/mnt/libvirt/vm_one/mydiskofpoolone_temporary" pid=16620 comm="qemu-system-x86" requested_mask="wr" denied_mask="wr" fsuid=64055 ouid=64055

First we only apply that to ubuntu-2004, and second we only match on `denied_mask="r"`, so we need to widen and lengthen it. But if most everything else succeeds, I propose to land this and fix this in a follow-up.

**Comment by martinpitt** [comment]
I sent PR #13745 to fix the ubuntu-stable messages.

**Pull request merged**